File: Server/Models/CarModel.cs
using System.ComponentModel.DataAnnotations;

namespace CarVerzel.Models;

public class Carro
{
    [Key]
    public int CarId { get; set; }

    [Required]
    public string Modelo { get; set; } = string.Empty;

    [Required]
    public string Marca { get; set; } = string.Empty;

    [Required]
    public decimal Preco { get; set; }

    [Required]
    
    public string FotoBase64 { get; set; } = string.Empty;
}

File: mobile/lib/src/Models/car_model.dart
class CarModel {
  int id;
  String modelo;
  String marca;
  double preco;
  String foto;

  CarModel({
    required this.id,
    required this.modelo,
    required this.marca,
    required this.preco,
    required this.foto,
  });

  factory CarModel.fromJson(Map<String, dynamic> json) {
    return CarModel(
      id: json['carId'],
      modelo: json['modelo'],
      marca: json['marca'],
      preco: json['preco'].toDouble(),
      foto: json['fotoBase64'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'carId': id,
      'modelo': modelo,
      'marca': marca,''
      'preco': preco,
      'fotoBase64': foto,
    };
  }
}
